### PR TITLE
Edge cases

### PR DESCRIPTION
**Pull request by jcrugzz**
## Summary
This fixes a subtle bug where we assumed the version being published in our test cases wasnt already in npm which was a silly and false assumption

## Changelog
* Fix subtle bug in edge case handling to exclude the version being published

## Test Plan
updated tests and they pass

### CODE REVIEW

File: lib/feedsme/index.js [hunk 1]
```diff
@@ -535,10 +540,12 @@ Feedsme.prototype.increment = function inc({ publish, strategy, payload, head, p
...
     // which would mess up our whole system here. If our version fetched is
     // somehow greater than the current build head, we will increrment that
-    // version
+    // version only if its in the same major version line
     //
     const latestHead = this._latestHead(head);
-    if (!latest.version || semver.lte(latest.version, latestHead)) {
+    if (!latest.version ||
+      (semver.lte(latest.version, latestHead) // head > latest.version
+        && semver.major(latest.version) === semver.major(latestHead))) { // && same major version line
```
Comment (msluther) [inline lib/feedsme/index.js:548]: This logic change could use a test case

**Review by msluther — APPROVED** [review]
Seems right I believe

**Review by ywangii — APPROVED** [review]

File: lib/feedsme/index.js (hunk 1 above)
Comment (jcrugzz) [reply lib/feedsme/index.js:548]: will make an issue for it and defer for now
